source/common/grpc/google_async_client_impl.cc

```diff
@@ -167,7 +167,24 @@ GoogleAsyncStreamImpl::GoogleAsyncStreamImpl(GoogleAsyncClientImpl& parent,
       service_full_name_(service_full_name), method_name_(method_name), callbacks_(callbacks),
       options_(options), unused_stream_info_(Http::Protocol::Http2, dispatcher_.timeSource(),
                                              Network::ConnectionInfoProviderSharedPtr{},
-                                             StreamInfo::FilterState::LifeSpan::FilterChain) {}
+                                             StreamInfo::FilterState::LifeSpan::FilterChain) {
+  if (nullptr != options.parent_span_) {
+    const std::string child_span_name =
+        options.child_span_name_.empty()
+            ? absl::StrCat("async ", service_full_name, ".", method_name, " egress")
```

Comment:
Where does this naming scheme come from, e.g. why is this "egress"?

Reply:
I used the same as unary grpc calls were using [[1]](https://github.com/envoyproxy/envoy/blob/99bb5659f8b204068cdb307a898d617d0ee4b5e8/source/common/grpc/google_async_client_impl.cc#L447-L449)[[2]](https://github.com/envoyproxy/envoy/blob/99bb5659f8b204068cdb307a898d617d0ee4b5e8/source/common/grpc/async_client_impl.cc#L301-L303) and also what was added in envoy grpc stream [[3]](https://github.com/envoyproxy/envoy/blob/99bb5659f8b204068cdb307a898d617d0ee4b5e8/source/common/grpc/async_client_impl.cc#L92-L95), let me know if there are guidelines to follow here.